pkg/storage/fs/ocis/grants.go

```diff
@@ -0,0 +1,380 @@
+		Permissions: permissions,
+		// TODO creator ...
+		Type: "A",
+	}
+	if g.Grantee.Type == provider.GranteeType_GRANTEE_TYPE_GROUP {
+		e.Flags = "g"
+	}
+	return e, nil
+}
+
+type ace struct {
+	//NFSv4 acls
+	Type        string // t
+	Flags       string // f
+	Principal   string // im key
+	Permissions string // p
+
+	// sharing specific
+	ShareTime int    // s
+	Creator   string // c
+	Expires   int    // e
+	Password  string // w passWord TODO h = hash
+	Label     string // l
+}
+
+func unmarshalACE(v []byte) (*ace, error) {
+	// first byte indicates type of value
+	switch v[0] {
+	case 0: // = ':' separated key=value pairs
+		s := string(v[1:])
+		return unmarshalKV(s)
+	default:
+		return nil, fmt.Errorf("unknown ace encoding")
+	}
+}
+
+func unmarshalKV(s string) (*ace, error) {
+	e := &ace{}
+	r := csv.NewReader(strings.NewReader(s))
+	r.Comma = ':'
+	r.Comment = 0
+	r.FieldsPerRecord = -1
+	r.LazyQuotes = false
+	r.TrimLeadingSpace = false
+	records, err := r.ReadAll()
+	if err != nil {
+		return nil, err
+	}
+	if len(records) != 1 {
+		return nil, fmt.Errorf("more than one row of ace kvs")
+	}
+	for i := range records[0] {
+		kv := strings.Split(records[0][i], "=")
+		switch kv[0] {
+		case "t":
+			e.Type = kv[1]
+		case "f":
+			e.Flags = kv[1]
+		case "p":
+			e.Permissions = kv[1]
+		case "s":
+			v, err := strconv.Atoi(kv[1])
+			if err != nil {
+				return nil, err
+			}
+			e.ShareTime = v
+		case "c":
+			e.Creator = kv[1]
+		case "e":
+			v, err := strconv.Atoi(kv[1])
+			if err != nil {
+				return nil, err
+			}
+			e.Expires = v
+		case "w":
+			e.Password = kv[1]
+		case "l":
+			e.Label = kv[1]
+			// TODO default ... log unknown keys? or add as opaque? hm we need that for tagged shares ...
+		}
+	}
+	return e, nil
+}
+
+// Parse parses an acl string with the given delimiter (LongTextForm or ShortTextForm)
+func getACEs(ctx context.Context, fsfn string, attrs []string) (entries []*ace, err error) {
+	log := appctx.GetLogger(ctx)
+	entries = []*ace{}
+	for i := range attrs {
+		if strings.HasPrefix(attrs[i], sharePrefix) {
+			principal := attrs[i][len(sharePrefix):]
+			var value []byte
+			if value, err = xattr.Get(fsfn, attrs[i]); err != nil {
+				log.Error().Err(err).Str("attr", attrs[i]).Msg("could not read attribute")
+				continue
+			}
+			var e *ace
+			if e, err = unmarshalACE(value); err != nil {
+				log.Error().Err(err).Str("attr", attrs[i]).Msg("could unmarshal ace")
+				continue
+			}
+			e.Principal = principal[2:]
+			// check consistency of Flags and principal type
+			if strings.Contains(e.Flags, "g") {
+				if principal[:1] != "g" {
+					log.Error().Str("attr", attrs[i]).Interface("ace", e).Msg("inconsistent ace: expected group")
+					continue
+				}
+			} else {
+				if principal[:1] != "u" {
+					log.Error().Str("attr", attrs[i]).Interface("ace", e).Msg("inconsistent ace: expected user")
+					continue
+				}
+			}
+			entries = append(entries, e)
+		}
+	}
+	return entries, nil
```

Comment:
true, we are logging the error ... but trying to continue to at least return normal metadata ... I don't think we should fail hard here, because the user cannot do anything about it. If a share does not show up he might retry creating the share ... hm

Reply:
removing the error